### ISSUE TITLE
Add shared web app styles imports

### ISSUE THREAD

**Issue by wlthomson** [openmsupply/unified-codes#186]
Minor continuous refactoring to lift out codebase styles from local components.

## Expected Behaviour
Web app UI is defined as DRY as possible, single style folder for all app styles to encourage consistency/easy maintenance.

## Current Behaviour
Styles defined at component level.

## Possible Solution
- Define web components with common `classes` + `styles` props and drill down.
- Export local components with material UI `styled` wrapper.
- Use material UI styles hooks (`makeStyles`).

Personally, I think the **second** option is the best for local components, as it cleans up the component props a bit and means we can focus the code on logic (I think then can have `style` and `class` but add explicit `component` props on library components, so don't have to mess around with coupling the library components to expected class titles etc. or drilling down nested styles... instead can just use the `styled` wrapped to pass components directly into the parent!). Personally, I don't think the `makeStyles` hook adds anything... I think it still requires hardcoding styles into components, which I think defeats the purpose a bit (have seen a lot of libraries adding hooks which don't seem to help that much... no expert, but in my humble opinion, there's a bit of a bandwagon thing going on since they've been introduced)!

**Comment by wlthomson** [comment]
@mark-prins @katherine-sussol Will do a WIP PR for this to: 

1. Estimate how much time it will take, may not be worth it at this stage.
2. Demo what I'm proposing before moving ahead with it.

**Comment by wlthomson** [comment]
OK, so have done a bit more research and have come to the following conclusion:

- `withStyles` is nice! Add a `classes` prop to app and library components (typed, so the API is clear), and you've got a flexible styling solution.
- Material UI themes look good, but so far I've found the API rigid and difficult (e.g. of this is @mark-prins having to override the `palette` to get flexibility for header and footer backgrounds). For a big app they may be useful... but for us, I think they will slow things down and come with more cons than pros. I can't see much that injecting common styles with `getStyles = (theme) => { body: theme.typography.body1 }` provides over statically importing styles and just doing `getStyles = () => { body: styles.body1 }` etc.
- It may be that we need themes just for the baseline material UI styles. If that's the case, I think we can using themes, but ideally keep it to a minimum?

@mark-prins You've done the existing implementation, anything I'm missing with this interpretation?

**Comment by wlthomson** [comment]
A few more questions, been doing more research!

Looks like the advantage of themes is that material UI takes a lot of the work out of stuff by automagically generating "lights", "darks" etc. from a user-defined palette. This only requires defining the following colours:

- `primary`
- `secondary`
- `error`
- `warning` 
- `info`
- `success`

I'm cool with the existing user-defined stuff, but it could be making more work for us than needed? Do we think the designer might be able to provide us with the above colours? 

If not, then do we agree with dispensing of material UI themes and just using `withStyles` directly?

**Comment by mark-prins** [comment]
Q1: we have a palette ( in inVision ) so we could plug that in and have a go. I suspect that we won't get very close to Pete's designs, so may have to wing it a bit. Probably worth it though

Q2: that's mainly due to inVision and how it represents colours. And it is the alpha channel that requires `rgba` - fine with converting, I had just left that up to the refactor

Generally - material UI theming is good and provides some benefit, as all their components support the standard theme. Define once and it all just works. Problem is that it isn't very flexible ( as I've demonstrated! ) and if you want to go off-piste there's a bit of mangling required.

I did try to use the standard theme shape and functionality, but couldn't bend it to implement the design as-is. The compromise is probably to modify the design, as you say.
If that doesn't look ok, then I we either extend the theme, as I've done, or abandon and work with a custom solution. In which case we may as well use semantic-ui, perhaps.
